edivorce/apps/core/efilinghub.py

```
@@ -0,0 +1,241 @@
+        logging.debug('EFH - Get Token {}'.format(response.status_code))
+        if response.status_code == 200:
+            response = json.loads(response.text)
+
+            # save in session .. lets just assume that current user is authenticated
+            if 'access_token' in response:
+                request.session['access_token'] = response['access_token']
+                if 'refresh_token' in response:
+                    request.session['refresh_token'] = response['refresh_token']
+
+                return True
+        return False
+
+    def _refresh_token(self, request):
+        refresh_token = request.session.get('refresh_token', None)
+        if not refresh_token:
+            return False
+
+        payload = 'client_id={}&grant_type=refresh_token&client_secret={}&refresh_token={}'.format(
+            self.client_id,
+            self.client_secret,
+            refresh_token)
+        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
+
+        url = '{}/auth/realms/{}/protocol/openid-connect/token'.format(self.token_base_url, self.token_realm)
+
+        response = requests.post(url, headers=headers, data=payload)
+        logging.debug('EFH - Get Refresh Token {}'.format(response.status_code))
+
+        response = json.loads(response.text)
+
+        # save in session .. lets just assume that current user is authenticated
+        if 'access_token' in response:
+            request.session['access_token'] = response['access_token']
+            if 'refresh_token' in response:
+                request.session['refresh_token'] = response['refresh_token']
+
+            return True
+        return False
+
+    def _get_api(self, request, url, transaction_id, bce_id, headers, data=None, files=None):
+        # make sure we have a session
+        access_token = request.session.get('access_token', None)
+        if not access_token:
+            if not self._get_token(request):
+                return False
```

Maybe you should raise an error here so the error is more obvious